x-pack/plugins/licensing/server/services/feature_usage_service.ts

```diff
@@ -43,14 +43,20 @@ export class FeatureUsageService {
   public setup(): FeatureUsageServiceSetup {
     return {
       register: (featureName, licenseType) => {
-        if (this.lastUsages.has(featureName)) {
-          throw new Error(`Feature '${featureName}' has already been registered.`);
+        const registered = this.lastUsages.get(featureName);
+        if (registered) {
+          if (registered.licenseType !== licenseType) {
+            throw new Error(
+              `Feature '${featureName}' has already been registered with another license type. (current: ${registered.licenseType}, new: ${licenseType})`
+            );
+          }
+        } else {
```

Comment:
@restrry This is the easiest fix for the problem, and answer the need.

Another option would be to instead adapt the client-side service to fetch the list of already registered features from the server-side during `setup` and to not perform calls to the server when trying to register an already existing feature. This is just a little trickier if we want to keep the client-side service synchronous

https://github.com/elastic/kibana/blob/5f781dcfecc9bcd96d9fa5ecfe5cd3e1cdedf094/x-pack/plugins/licensing/public/services/feature_usage_service.ts#L41-L53

as it would mean creating the fetch promise during `setup` and await for its result in every call to register. But nothing that can't be done. Do you think this would be better?

Reply:
I like the solution you have in PR. It's simpler and less likely to break IMO.